features/netsocket/emac-drivers/TARGET_RDA_EMAC/RdaWiFiInterface.cpp

```diff
@@ -0,0 +1,260 @@
+}
+
+int8_t RDAWiFiInterface::get_rssi()
+{
+    return rda5981_get_rssi();
+}
+
+nsapi_error_t RDAWiFiInterface::init()
+{
+    if (!_interface) {
+        if (!_emac.power_up()) {
+            LWIP_DEBUGF(NETIF_DEBUG,"power up failed!\n");
+        }
+        nsapi_error_t err = _stack.add_ethernet_interface(_emac, true, &_interface);
+        if (err != NSAPI_ERROR_OK) {
+            _interface = NULL;
+            return err;
+        }
+        _interface->attach(_connection_status_cb);
+    }
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t RDAWiFiInterface::set_credentials(const char *ssid, const char *pass,
+                nsapi_security_t security)
+{
+    if(ssid == 0 || strlen(ssid) == 0)
+        return NSAPI_ERROR_PARAMETER;
+    if(security != NSAPI_SECURITY_NONE && (pass == 0 || strlen(pass) == 0))
+        return NSAPI_ERROR_PARAMETER;
+    if(strlen(ssid) > 32 || strlen(pass) > 63)
+        return NSAPI_ERROR_PARAMETER;
+    memcpy((void*)_ssid, (void*)ssid, strlen(ssid));
+    _ssid[strlen(ssid)] = '\0';
+    memcpy((void*)_pass, (void*)pass, strlen(pass));
+    _pass[strlen(pass)] = '\0';
+    _security = security;
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t RDAWiFiInterface::connect(const char *ssid, const char *pass,
+                nsapi_security_t security, uint8_t channel)
+{
+    rda_msg msg;
+    bool find = false;
+    int i = 0;
+    rda5981_scan_result bss;
+    int ret = 0;
+
+    if (ssid == NULL || ssid[0] == 0) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+
+    init();
+
+    if(rda5981_check_scan_result_name(ssid) != 0) {
+        for (i = 0; i< 5; i++) {
+            rda5981_scan(NULL, 0, 0);
+            if(rda5981_check_scan_result_name(ssid) == 0) {
+                find = true;
+                break;
+            }
+        }
+    } else {
+        find = true;
+    }
+
+    if (find == false) {
+        LWIP_DEBUGF(NETIF_DEBUG,"can not find the ap.\r\n");
+        return NSAPI_ERROR_CONNECTION_TIMEOUT;
+    }
+    bss.channel = 15;
+    rda5981_get_scan_result_name(&bss, ssid);
+    if ((channel !=0) && (bss.channel != channel)) {
+        LWIP_DEBUGF(NETIF_DEBUG, "invalid channel\r\n");
+        return NSAPI_ERROR_CONNECTION_TIMEOUT;
+    }
+
+    memcpy(gssid, ssid, strlen(ssid));
+    if(pass[0] != 0)
+        memcpy(gpass, pass, strlen(pass));
+    memset(gbssid, 0, NSAPI_MAC_BYTES);
+    gssid[strlen(ssid)] = gpass[strlen(pass)] = '\0';
+
+    msg.type = WLAND_CONNECT;
+    rda_mail_put(wland_msgQ, (void*)&msg, osWaitForever);
+    ret = rda_sem_wait(wifi_auth_sem, 10000);
+    if (ret) {
+        return NSAPI_ERROR_CONNECTION_TIMEOUT;
+    }
+
+    ret = _interface->bringup(_dhcp,
+          _ip_address[0] ? _ip_address : 0,
+          _netmask[0] ? _netmask : 0,
+          _gateway[0] ? _gateway : 0,
+          DEFAULT_STACK,
+          _blocking);
+
+    return ret;
+}
+
+
+nsapi_error_t RDAWiFiInterface::connect()
+{
+#if 1
+    return connect(_ssid, _pass, _security, _channel);
+#else
```

Comment:
In the future, you might want to remove the dead code from the driver.
When left there, it soon gets outdated, and might be misleading for the next developer who tries to understand this driver.

Reply:
you are right, and I have remove the dead code just now.